### PR TITLE
Migrate SampleConditions to Dexterity

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses

This PR migrates the AT SampleConditions to Dexterity types for Python 3 compatibility.

## Current behavior before PR

SampleConditions are based on the Archetypes framework

## Desired behavior after PR is merged

SampleConditions are based on the Dexterity framework

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/senaite/core/browser/controlpanel/sampleconditions/view.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import collections
+
+from bika.lims import bikaMessageFactory as _
+from bika.lims.browser.bika_listing import BikaListingView
+from bika.lims.utils import get_link_for
+from senaite.core.catalog import SETUP_CATALOG
+from senaite.core.permissions import AddSampleCondition
+
+
+class SampleConditionsView(BikaListingView):
```
Comment (ramonski) [inline src/senaite/core/browser/controlpanel/sampleconditions/view.py]: Can we import here directly from `senaite.app.listing`?

File: src/senaite/core/browser/controlpanel/sampleconditions/view.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    def __init__(self, context, request):
+        super(SampleConditionsView, self).__init__(context, request)
+
+        self.catalog = SETUP_CATALOG
+        self.show_select_row = False
```
Comment (ramonski) [inline src/senaite/core/browser/controlpanel/sampleconditions/view.py]: This option does no longer exist

File: src/senaite/core/upgrade/v02_06_000.py [hunk 3]
```diff
@@ -93,26 +139,64 @@ def migrate_departments_to_dx(tool):
...
+
+    # migrate the contents from the old AT container to the new one
+    migrate_to_dx("SampleCondition", origin, destination, schema_mapping)
+
+    # copy snapshots for the container
+    copy_snapshots(origin, destination)
+
+    # remove old AT folder
+    if len(origin) == 0:
+        delete_object(origin)
```
Comment (ramonski) [inline src/senaite/core/upgrade/v02_06_000.py:166]: Why this test for `len(origin)`?

Comment (xispa) [reply src/senaite/core/upgrade/v02_06_000.py:166]: For e.g. SampleConditions we do know the folder will contain objects from SanpleCondition type. We cannot say the same for other folders we might migrate objects from in future (like e.g. Client when migrating SampleType objects)

Comment (ramonski) [reply src/senaite/core/upgrade/v02_06_000.py:166]: But this specific line is in the function `migrate_sampleconditions_to_dx`. So it will not be deleted?

Comment (ramonski) [reply src/senaite/core/upgrade/v02_06_000.py:166]: Ah, now I got it. You are deleting it explicitly in `migrate_to_dx`:
```
        # Migrate the contents from AT to DX
        migrator = getMultiAdapter(
            (src, target), interface=IContentMigrator)
        migrator.migrate(schema_mapping, delete_src=False)

        # Delete the old AT object
        delete_object(src)
```

Please change the code above and pass in `delete_src` to the migrator instead.

File: src/senaite/core/upgrade/v02_06_000.py [hunk 4]
```diff
@@ -80,11 +82,55 @@ def remove_at_portal_types(tool):
...
+            dest._setObject(src_id, target)
+            target = dest._getOb(src_id)
+
+        # Migrate the contents from AT to DX
+        migrator = getMultiAdapter(
+            (src, target), interface=IContentMigrator)
+        migrator.migrate(schema_mapping, delete_src=False)
+
+        # Delete the old AT object
+        delete_object(src)
```
Comment (ramonski) [inline src/senaite/core/upgrade/v02_06_000.py]: Please remove this line and pass in `delete_src=True` to the migrator call above

**Review by ramonski — APPROVED** [review]
One step closer to the Goal! Thanks!

**Pull request merged**